packages/telemetry/index.js

```diff
@@ -0,0 +1,155 @@
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const { name: moduleName, version: moduleVersion } = require('./package.json')
+
+function formatSpanName (request) {
+  const { method, routerPath } = request
+  /* istanbul ignore next */
+  return routerPath ? `${method} ${routerPath}` : method
+}
+
+const defaultFormatSpanAttributes = {
+  request (request) {
+    return {
+      'req.method': request.raw.method,
+      'req.url': request.raw.url
+    }
+  },
+  reply (reply) {
+    return {
+      'reply.statusCode': reply.statusCode
+    }
+  },
+  error (error) {
+    return {
+      'error.name': error.name,
+      'error.message': error.message,
+      'error.stack': error.stack
+    }
+  }
+}
+
+const setupProvider = (app, opts) => {
+  const { serviceName, version } = opts
+  let exporter = opts.exporter
+  if (!exporter) {
+    app.log.warn('No exporter configured, defaulting to console.')
+    exporter = { type: 'console' }
+  }
+  app.log.info(`Setting up telemetry for service: ${serviceName} version: ${version} and exporter of type ${exporter.type}`)
+  const provider = new PlatformaticTracerProvider({
+    resource: new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: serviceName,
+      [SemanticResourceAttributes.SERVICE_VERSION]: version
+    })
+  })
+
+  // Exporter config:
+  // https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_exporter_zipkin.ExporterConfig.html
+  const exporterOptions = { ...exporter.options, serviceName }
+  let exporterObj
+  switch (exporter.type) {
+    case 'console':
+      exporterObj = new ConsoleSpanExporter(exporterOptions)
+      break
+    case 'otlp':
+      exporterObj = new OTLPTraceExporter(exporterOptions)
+      break
+    case 'zipkin':
+      exporterObj = new ZipkinExporter(exporterOptions)
+      break
+    case 'memory':
+      // This is for testing
+      exporterObj = new InMemorySpanExporter()
+      break
+    default:
+      app.log.warn(`Unknown exporter type: ${exporter.type}, defaulting to console.`)
+      exporterObj = new ConsoleSpanExporter(exporterOptions)
+  }
+
+  // We use a SimpleSpanProcessor for the console/memory exporters and a BatchSpanProcessor for the others.
+  const spanProcessor = ['memory', 'console'].includes(exporter.type) ? new SimpleSpanProcessor(exporterObj) : new BatchSpanProcessor(exporterObj)
+  provider.addSpanProcessor(spanProcessor)
+  const tracer = provider.getTracer(moduleName, moduleVersion)
```

Comment:
I'm not sure we want to use `moduleName` `moduleVersion` here

Reply:
This is by spec: https://opentelemetry.io/docs/specs/otel/trace/api/#get-a-tracer